Dockerfile

```diff
@@ -1,31 +1,52 @@
-FROM python:3.7-slim-stretch
+# First build the Helm binding
+FROM golang:1.12.9-stretch AS go-builder
 
 RUN mkdir /kapitan
 WORKDIR /kapitan
-COPY kapitan/ kapitan/
-COPY requirements.txt ./
+COPY . .
 
-RUN apt-get update && apt-get install -y \
-    build-essential git wget \
-    && apt-get clean \
-    && rm -rf /var/lib/apt/lists/* \
-    && pip install --upgrade pip \
-    && pip install -r requirements.txt
+RUN ./kapitan/inputs/helm/build.sh
 
-# build go from source
-RUN wget https://dl.google.com/go/go1.12.7.linux-amd64.tar.gz -q && \
-    tar -C /usr/local -xvf go1.12.7.linux-amd64.tar.gz && \
-    rm go1.12.7.linux-amd64.tar.gz
+# Build the virtualenv for Kapitan
+FROM python:3.7-slim-stretch AS python-builder
 
-# build helm binding
-RUN export PATH=$PATH:/usr/local/go/bin && \
-    chmod +x kapitan/inputs/helm/build.sh && \
-    ./kapitan/inputs/helm/build.sh && \
-    rm /usr/local/go -rf
+COPY --from=go-builder /kapitan /kapitan
+WORKDIR /kapitan
 
-ENV PYTHONPATH="/kapitan/"
+ENV PATH="/opt/venv/bin:${PATH}"
+
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+        build-essential \
+        gcc \
+    && python -m venv /opt/venv \
+    && pip install --upgrade \
+        pip \
+        setuptools \
+        wheel \
+    && pip install -r requirements.txt \
+    && ./kapitan/inputs/helm/build.sh \
```

Comment:
@uberspot 
Let me explain because I wrote this script. `build.sh` creates two files, `libtemplate.so` for which we need go, and `helm_binding.py` (which we need to dlopen `libtemplate.so`) for which we need python3. go container doesn't have python3, so we initially create only `libtemplate.so` in the go container, and then copy that over to python container to generate `helm_binding.py` using the same script.

Reply:
Got it, thanks for explaining. :) 
Would be good to have a short comment above it to avoid future confusion. 
I'll resolve this.